src/contract.rs

```diff
@@ -765,9 +832,28 @@ fn try_burn_from<S: Storage, A: Api, Q: Querier>(
     owner: &HumanAddr,
     amount: Uint128,
 ) -> StdResult<HandleResponse> {
+    let mut config = Config::from_storage(&mut deps.storage);
+    let constants = config.constants()?;
+    if !constants.burn_is_enabled {
+        return Err(StdError::generic_err(
+            "Burn functionality is not enabled for this token.",
+        ));
+    }
+
+    // remove from supply
+    let amount = amount.u128();
+    let mut total_supply = config.total_supply();
+    if let Some(new_total_supply) = total_supply.checked_sub(amount) {
+        total_supply = new_total_supply;
+    } else {
+        return Err(StdError::generic_err(
+            "You're trying to burn more than is available in the total supply",
+        ));
+    }
+    config.set_total_supply(total_supply);
```

Comment:
like in redeem, we should check the user's balance before checking the total supply

Reply:
The redeem comment reply applies here as well